ui/pages/confirmations/hooks/alerts/transactions/useResimulationAlert.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { RowAlertKey } from '../../../../../components/app/confirm/info/row/constants';
+import { useConfirmContext } from '../../../context/confirm';
+
+export function useResimulationAlert(): Alert[] {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext();
+
+  if (!currentConfirmation) {
+    return [];
+  }
+
+  const { simulationData } = currentConfirmation;
+
+  if (!simulationData) {
+    return [];
```

Comment:
We can't early return before the `useMemo` since it's a hook, but we can instead just do optional chaining to check the property?

Reply:
Done